nexus/transport.go

```diff
@@ -144,6 +183,33 @@ func (r *OperationResult[T]) Get() (T, error) {
 	return r.result, r.err
 }
 
+// UnimplementedTransport must be embedded into any [Transport] implementation for future compatibility.
+// It implements all methods on the [Transport] interface, returning unimplemented errors if they are not implemented by
+// the embedding type.
+type UnimplementedTransport struct{}
+
+func (u UnimplementedTransport) mustEmbedUnimplementedTransport() {}
+
+func (u UnimplementedTransport) StartOperation(_ context.Context, _ any, _ TransportStartOperationOptions) (*TransportStartOperationResponse[*LazyValue], error) {
+	return nil, HandlerErrorf(HandlerErrorTypeNotImplemented, "not implemented")
```

Comment:
This should be a transport error IMHO:

```suggestion
	return nil, &TransportError{
	  Message: "not implemented",
	}
```

Reply:
Whoops. Addressed now.